tests/Feature/SlotAssignmentTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+use App\Helpers\SlotAssignment;
+use App\Models\Event;
+use App\Models\Slot;
+use App\Models\User;
+use App\Models\Course;
+use App\Models\Registration;
+
+uses(\Illuminate\Foundation\Testing\RefreshDatabase::class);
+
+beforeEach(function () {
+    // create a new event
+    $this->event = new Event();
+    $this->event->name = 'slot_booking_test';
+    $this->event->description = 'Lorem ipsum';
+    $this->event->type = 'slot_booking';
+
+    $registration_from = new DateTime(NOW());
+    $registration_to = new DateTime(NOW());
+    $registration_to->modify('+7 days');
+
+    $this->event->registration_from = $registration_from;
+    $this->event->registration_to = $registration_to;
+    $this->event->has_requirements = false;
+    $this->event->consider_alcohol = false;
+    $this->event->sort_order = 300;
+    $this->event->save();
+
+    // cerate a new slot
+    $this->slot = new Slot();
+    $this->slot->name = "small_test";
+    $this->slot->event_id = $this->event->id;
+    $this->slot->has_requirements = false;
+    $this->slot->maximum_participants = "10";
+    $this->slot->save();
+
+    Course::factory()->count(1)->create();
+
+    $this->assignment = new SlotAssignment($this->slot);
+});
+
+function createUsers(int $amount)
+{
+    return User::factory()
+        ->has(
+            Registration::factory()
+                ->state([
```

Comment:
Why was it decided here to use the `state()` method for an inline state transformation?

Reply:
@jpih has to answer that